bundles/binding/org.openhab.binding.ddwrt/src/main/java/org/openhab/binding/ddwrt/internal/DDWRTBinding.java

```diff
@@ -0,0 +1,483 @@
+import org.openhab.core.items.Item;
+import org.openhab.core.library.items.NumberItem;
+import org.openhab.core.library.items.StringItem;
+import org.openhab.core.library.items.SwitchItem;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.osgi.service.cm.ConfigurationException;
+import org.osgi.service.cm.ManagedService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The DD-WRT binding connects to a DD-WRT router and switch the WIFI devices.
+ *
+ * @author Markus Eckhardt
+ * @author Kai Kreuzer
+ * @author Jan N. Klug
+ *
+ * @since 1.9.0
+ */
+public class DDWRTBinding extends AbstractActiveBinding<DDWRTBindingProvider> implements ManagedService {
+
+    private static HashMap<String, String> queryMap = new HashMap<>();
+
+    static {
+        queryMap.put(DDWRTBindingProvider.TYPE_ROUTER_TYPE, "nvram get DD_BOARD");
+        queryMap.put(DDWRTBindingProvider.TYPE_WLAN_24, "ifconfig");
+        queryMap.put(DDWRTBindingProvider.TYPE_WLAN_50, "ifconfig");
+        queryMap.put(DDWRTBindingProvider.TYPE_WLAN_GUEST, "ifconfig");
+    }
+
+    @Override
+    public void bindingChanged(BindingProvider provider, String itemName) {
+        super.bindingChanged(provider, itemName);
+
+        conditionalDeActivate();
+
+    }
+
+    private void conditionalDeActivate() {
+        logger.trace("DD-WRT conditional deActivate: {}", bindingsExist());
+
+        if (bindingsExist()) {
+            activate();
+        } else {
+            deactivate();
+        }
+    }
+
+    private static final Logger logger = LoggerFactory.getLogger(DDWRTBinding.class);
+
+    /* The IP address to connect to */
+    protected static String ip;
+
+    /* The port to connect to, default 23 */
+    protected static String port;
+
+    /* The password of the DD-WRT to access via Telnet */
+    protected static String password;
+
+    /* The username, if used for telnet connections */
+    protected static String username;
+
+    /* The interface for 2.4 GHz network, e.g. ath0 */
+    protected static String interface_24;
+
+    /* The interface for 5.0 GHz network, e.g. ath1 */
+    protected static String interface_50;
+
+    /* The interface for guest network, e.g. virt_ath0 */
+    protected static String interface_guest;
+
+    @Override
+    public void activate() {
+        super.activate();
+        setProperlyConfigured(true);
+        logger.debug("DD-WRT binding has been started.");
+    }
+
+    @Override
+    public void deactivate() {
+        super.deactivate();
+        logger.debug("DD-WRT binding has been started.");
+    }
+
+    @Override
+    public void internalReceiveCommand(String itemName, Command command) {
+
+        logger.trace("internalReceiveCommand");
+        if (password != null && StringUtils.isNotBlank(password)) {
```

Comment:
The `null` check isn't needed (see [here](https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/StringUtils.html#isNotBlank(java.lang.String))).